main/src/ca/uwaterloo/flix/language/phase/CodegenHelper.scala

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2015-2017 Ramin Zarifi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api._
+import ca.uwaterloo.flix.language.GenSym
+import ca.uwaterloo.flix.language.ast.ExecutableAst.{Definition, Expression}
+import ca.uwaterloo.flix.language.ast.Symbol.EnumSym
+import ca.uwaterloo.flix.language.ast.{Type, _}
+import ca.uwaterloo.flix.runtime.Value
+import ca.uwaterloo.flix.util.InternalCompilerException
+import org.objectweb.asm
+import org.objectweb.asm.Opcodes._
+import org.objectweb.asm.{Type => _, _}
+
+object CodegenHelper {
+
+  val fixedPrefix = List("ca", "waterloo", "flix", "enums")
+
+  case class QualName(ref: List[String])
+
+  def getEnumCaseName(sym: EnumSym, tag: String, tpe: Option[Type]) : QualName = tpe match {
+    case Some(t) if t.isPrimitive => QualName(CodegenHelper.fixedPrefix ++ sym.namespace ++ List(sym.name, t.toString, tag))
+    case _ => QualName(CodegenHelper.fixedPrefix ++ sym.namespace ++ List(sym.name, "object", tag))
+  }
+
+  def getEnumInterfaceName(sym: EnumSym): QualName = {
+    QualName(CodegenHelper.fixedPrefix ++ sym.namespace ++ List(sym.name, "EnumInterface"))
+  }
+
+  /*
+   * Decorate (mangle) a prefix (list of strings) to get the internal JVM name.
+   */
+  def decorate(prefix: List[String]): String = prefix.mkString("/")
+
+  def decorate(qualName: QualName): String = decorate(qualName.ref)
+
+  /*
+   * Returns the internal name of the JVM type that `tpe` maps to.
```

Comment:
The comment still applies, I didn't change the function, I just moved a variable in the case class to be directly given to the function. I changed the format of the comment though so it looks like the rest.